tests/test_bot.py

```diff
@@ -116,6 +116,22 @@ def test_detect_self_bump(self, _) -> None:
 
         self.cleanup()
 
+    def test_detect_url(self, _) -> None:
+        """Test the ability of KarmnaBot to detect a URL which contains either the ++ or
+        -- tokens.
+        """
+        cases = [
```

Comment:
Let me introduce you to the wonders of [Pytest's `parameterize` mark](https://docs.pytest.org/en/6.2.x/parametrize.html#pytest-mark-parametrize)!